source/common/http/filter_manager.cc

```diff
@@ -538,7 +536,8 @@ void FilterManager::decodeHeaders(ActiveStreamDecoderFilter* filter, RequestHead
       addDecodedData(*((*entry).get()), empty_data, true);
     }
 
-    if (!continue_iteration && std::next(entry) != decoder_filters_.end()) {
+    if ((!continue_iteration || state_.local_complete_) &&
+        std::next(entry) != decoder_filters_.end()) {
```

Comment:
Do we have an integration test with headers-and-body with a test filter which sends local reply on headers?  It'd be good to make sure we verify the rest of the filter chain handles this gracefully.

Reply:
Test added, PTAL. I'm not sure if it covers the filter chain handling that you had in mind.

Reply:
yeah, I wanted a test which would have previously run this loop with local_complete_ == true, which I believe this does.